src/weapon/w_058.c

```diff
@@ -17,7 +19,196 @@ s32 func_ptr_80170014(Entity* self) {}
+void EntityWeaponShieldSpell(Entity* self) {
+    Primitive* prim;
+    s16 primRight;
+    s16 primLeft;
+    s16 selfX;
+    s16 selfY;
+    bool showQuestionMark;
+
+    // This block sets variables that are unused
+    s16 temp_s1;
+    s16 temp_s5;
+
+    temp_s1 = 0x1C;
+    if (PLAYER.facingLeft) {
+        temp_s1 = -temp_s1;
+    }
+    temp_s5 = 0x30;
+
+    showQuestionMark = false;
+    switch (self->step) {
+    case 0:
+        self->primIndex = g_api.AllocPrimitives(PRIM_GT4, 2);
+        if (self->primIndex == -1) {
+            DestroyEntity(self);
+            return;
+        }
+        SetSpriteBank1(D_19A000_8017A040);
+        if (g_HandId != 0) {
+            self->animSet = ANIMSET_OVL(0x12);
+            self->palette = 0x128;
+            self->ext.shield.unk86 = 0x129;
+            self->unk5A = 0x66;
+            self->ext.shield.unk7C = 0;
+            self->ext.shield.unk7D = 0x80;
+        } else {
+            self->animSet = ANIMSET_OVL(0x10);
+            self->palette = 0x110;
+            self->ext.shield.unk86 = 0x111;
+            self->unk5A = 0x64;
+            self->ext.shield.unk7C = 0x80;
+            self->ext.shield.unk7D = 0;
+        }
+
+        self->posY.i.hi -= 8;
+        self->flags = FLAG_UNK_04000000 | FLAG_HAS_PRIMS | FLAG_UNK_10000;
+        self->zPriority = PLAYER.zPriority - 2;
+        self->facingLeft = PLAYER.facingLeft;
+        self->animCurFrame = 0x3E;
+        self->drawFlags = FLAG_DRAW_ROTX | FLAG_DRAW_ROTY;
+        self->rotX = self->rotY = 0;
+        // The primitive for the main shield appearing up behind Alucard
+        prim = &g_PrimBuf[self->primIndex];
+        prim->clut = 0x19F;
+        prim->tpage = 0x19;
+        prim->u0 = prim->u2 = 0x80;
+        prim->u1 = prim->u3 = 0x80 + 0x4F;
+        prim->v0 = prim->v1 =
+            self->ext.shield.unk7C + 0; // PSP needs +0 which is interesting
+        prim->v2 = prim->v3 = self->ext.shield.unk7C + 0x4F;
+        self->ext.shield.unk82 = 0;
+        prim->r0 = prim->g0 = prim->b0 = prim->r1 = prim->g1 = prim->b1 =
+            prim->r2 = prim->g2 = prim->b2 = prim->r3 = prim->g3 = prim->b3 =
+                0x80;
+        prim->priority = self->zPriority - 4;
+        prim->drawMode =
+            DRAW_UNK_40 | DRAW_TPAGE2 | DRAW_TPAGE | DRAW_COLORS | DRAW_TRANSP;
+        prim = prim->next;
+        // The primitive for the little question mark above Alucard
+        prim->clut = 0x190;
+        prim->tpage = 0x1E;
+        prim->u0 = prim->u2 = 0x78;
+        prim->u1 = prim->u3 = 0x78 + 7;
+        prim->v0 = prim->v1 = 8;
+        prim->v2 = prim->v3 = 8 + 7;
+        prim->priority = self->zPriority + 4;
+        prim->drawMode = DRAW_HIDE;
+        SetSpeedX(FIX(-2));
+        self->velocityY = 0;
+        DestroyEntityWeapon(1);
+        g_api.PlaySfx(SFX_UNK_641);
+        g_unkGraphicsStruct.unk20 = 1;
+        // Would be nice if this could be removed
```

Comment:
Now that you have removed the fake symbol in favour of `g_unkGraphicsStruct.unk20`, you should be able to remove the do while(0)

Reply:
Oh holy crap, you're totally right, awesome catch!